man/systemd-nspawn.xml

```diff
@@ -1352,6 +1352,58 @@ After=sys-subsystem-net-devices-ens1.device</programlisting>
         make them read-only, using <option>--bind-ro=</option>.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--bind-user=</option></term>
+
+        <listitem><para>Binds the home directory of the specified user on the host into the container. Takes
+        the name of an existing user on the host as argument. May be used multiple times to bind multiple
+        users into the container. This does three things:</para>
+
+        <orderedlist>
+          <listitem><para>The user's home directory is bind mounted from the host into
+          <filename>/run/hosts/home/</filename>.</para></listitem>
+
+          <listitem><para>An additional UID/GID mapping is added that maps the host user's UID/GID to a
+          container UID/GID, allocated from the 60514…60577 range.</para></listitem>
+
+          <listitem><para>A JSON user and group record is generated in <filename>/run/userdb/</filename> that
+          describes the the mapped user. It contains a minimized representation of the host's user record,
+          adjusted to the UID/GID and home directory path assigned to the user in the container. The
+          <citerefentry><refentrytitle>nss-systemd</refentrytitle><manvolnum>8</manvolnum></citerefentry>
+          glibc NSS module will pick up these records from there and make them available in the containers
+          user/group databases.</para></listitem>
+        </orderedlist>
+
+        <para>The combination of the three operations above ensures that it is possible to log into the
+        host's user account inside the container as if it was local to the container. The user is only mapped
+        transiently, while the container is running and the mapping itself does not result in persistent
+        changes to the container (except maybe for generated log messages at login time, and similar). Note
+        that in particular the UID/GID assignment in the container is not made persistently. Thus: if a user
+        after logging in makes changes to the image (outside of their home directory or transient
+        directories), and leaves files or directories around that are owned by them, they will remain
+        associated with the used UID/GID. As these UID/GIDs might be recycled during later container
+        invocations (with possibly different <option>--bind-user=</option> settings) and be assigned to
+        different users or groups, this is potentially a security issue.</para>
```

Comment:
Hmm, I also think this is unnecessarily constraining. Normal user users can create various semi-permanent files, e.g. in `/var/tmp`, or maybe with help of some local privilege escalation helpers. I think this is fine. Maybe say:
> If the user is mapped transiently, it is best to not allow the user to make persistent changes to the container. If the user leaves files or directories owned by that user, and those UIDs/GIDs are recycled during later container invocations (possibly with a different `<option>--bind-user=</option>` mapping), those files and directories will be accessible to the "new" user.